### PR TITLE
bitmex connection, book and trade tests

### PR DESCRIPTION
**Pull request by Foat**

### CODE REVIEW

File: xchange-stream-core/src/main/java/info/bitrich/xchangestream/util/PropsLoader.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * 08/06/2018
+ */
+public class PropsLoader {
+    private static final Logger LOG = LoggerFactory.getLogger(PropsLoader.class);
+
+    public static LocalExchangeConfig loadKeys(String fileName, String originName) throws IOException {
+        try (FileInputStream input = new FileInputStream(fileName)) {
+            Properties properties = new Properties();
+            properties.load(input);
+            return new LocalExchangeConfig(properties.getProperty("api.key"), properties.getProperty("secret.key"));
```
Comment (jkolobok) [inline xchange-stream-core/src/main/java/info/bitrich/xchangestream/util/PropsLoader.java]: I'd add an option to use -D or environment properties to get values

File: xchange-bitmex/src/test/java/info/bitrich/xchangestream/bitmex/BitmexOrderTest.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+        if (topPriceAsk != null) {
+            testAskPrice = topPriceAsk.add(priceShift);
+            testBidPrice = topPriceAsk.subtract(priceShift);
+        } else {
+            testAskPrice = topPriceBid.add(priceShift);
+            testBidPrice = topPriceBid.subtract(priceShift);
+        }
+
+        tradeService = (BitmexTradeService) exchange.getTradeService();
```
Comment (jkolobok) [inline xchange-bitmex/src/test/java/info/bitrich/xchangestream/bitmex/BitmexOrderTest.java:89]: There are some functions that are not available via BitmexTradeService and it's better to use BitmexTradeServiceRaw.
it has order replacement replace function.. and cancellAllOrders

File: xchange-bitmex/src/test/java/info/bitrich/xchangestream/bitmex/BitmexOrderTest.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        return System.currentTimeMillis() + "";
+    }
+
+    private String placeLimitOrder(String nosOrdId, BigDecimal price, String size, Order.OrderType type) throws Exception {
+        LimitOrder limitOrder =
+                new LimitOrder(
+                        type,
+                        new BigDecimal(size),
+                        xbtUsd,
+                        nosOrdId,
```
Comment (jkolobok) [inline xchange-bitmex/src/test/java/info/bitrich/xchangestream/bitmex/BitmexOrderTest.java]: so here is a note that in the limitorder orderid works as clOrdID...

File: xchange-bitmex/src/test/java/info/bitrich/xchangestream/bitmex/BitmexOrderTest.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+                        new Date(),
+                        price);
+        String orderId = tradeService.placeLimitOrder(limitOrder);
```
Comment (jkolobok) [inline xchange-bitmex/src/test/java/info/bitrich/xchangestream/bitmex/BitmexOrderTest.java]: And here it works as orderID

File: xchange-bitmex/src/test/java/info/bitrich/xchangestream/bitmex/BitmexOrderTest.java (hunk 2 above)
Comment (Foat) [reply xchange-bitmex/src/test/java/info/bitrich/xchangestream/bitmex/BitmexOrderTest.java:89]: BitmexTradeServiceRaw is just a superclass for BitmexTradeService, so all functions are available from BitmexTradeService

**Pull request merged**